src/fstab-generator/fstab-generator.c

```diff
@@ -1151,6 +1261,34 @@ static int add_volatile_var(void) {
                          SPECIAL_LOCAL_FS_TARGET);
 }
 
+static int add_mounts_from_cmdline(void) {
+        int r, ret = 0;
+        bool saved;
+
+        /* Handle each entries found in cmdline as a fstab entry. */
+
+        /* To make swap units specified in systemd.swap-extra= created, even if systemd.swap=no. */
+        saved = arg_swap_enabled;
+        arg_swap_enabled = true;
```

Comment:
this is a bit mean, maybe instead add another param to parse_fstab_one(), i.e. "honour_swap_enabled" or so that we set to false here, and true otherwise.

(or maybe even make this a proper flags param, together with the initrd boolean, but that'd just be the cherry on top)

Reply:
or actually, why even bother? i mean, if you use the kernel cmdline to disable swap and then configure swap there, is there really a strong reason to not just say "hey, you turned if off, so we'll ignore this now"?

Reply:
systemd.swap= controls if we should handle swap entry in fstab. I'd like to support `systemd.swap=no systemd.swap-extra=...` to replace swap entry in fstab with the one specified in command line.

Reply:
> this is a bit mean, maybe instead add another param to parse_fstab_one(), i.e. "honour_swap_enabled" or so that we set to false here, and true otherwise.

Done.